app/routes.py

```diff
@@ -1,2 +1,153 @@
+            db.session.commit()
+        return make_response({"id": goal.goal_id, "task_ids": request_body["task_ids"]}, 200)
+
+
+####################################################################
+#                              TASK ROUTES                        #
+####################################################################
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc": 
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+            
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append(task.task_data_structure()['task'])
+
+        return make_response(jsonify(tasks_response), 200)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if 'title' in request_body and 'description' in request_body and 'completed_at' in request_body: 
+            new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body['completed_at'])
+            db.session.add(new_task)
+            db.session.commit()
+            return make_response(new_task.task_data_structure(), 201)
+        else:
+            return make_response({"details":"Invalid data"}, 400)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if request.method == "GET":
+        return make_response(task.task_data_structure(), 200)
+    elif request.method == "PUT":
+        task_data = request.get_json()
+        task.title = task_data["title"]
+        task.description = task_data["description"]
+        task.completed_at = task_data['completed_at']
+        db.session.commit()
+        task = Task.query.get(task_id)
+        return make_response(task.task_data_structure(), 200)
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit() 
+        task_response = {"details": f'Task {task.task_id} "{task.title}" successfully deleted'}
+        return make_response(jsonify(task_response), 200)
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_complete(task_id):
+    task = Task.query.get_or_404(task_id)
+    task.update_completed() # function in Task Module to Update Time
+    db.session.commit() 
+    # uses Slack API to interact with a slackbot + post to channel
+    url = f"https://slack.com/api/chat.postMessage?channel=general&text={'Hello Testing'}"
+    auth_token = os.environ.get("API_KEY")
+    headers = {'Authorization': auth_token}
+    # No Payload as text is sent via query string
+    response = requests.request("POST", url, headers=headers)
```

Comment:
👍 heck yeah! this looks good! this would be a perfect opportunity to turn this into a separate helper function outside this route.